### PR TITLE
[INLONG-4790][SDK] Upgrade DES encryption decryption to AES

### LINKED ISSUE #4790: [Improve][DataProxySDK] Replace DES with a more secure encryption algorithm
### Description

https://github.com/apache/incubator-inlong/pull/4372/checks?check_run_id=6892208988
image

### InLong Component

InLong SDK

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by woofyzhao**
- Fixes #4790

### CODE REVIEW

File: inlong-sdk/dataproxy-sdk/src/main/java/org/apache/inlong/sdk/dataproxy/example/UdpClientExample.java [hunk 1]
```diff
@@ -92,6 +92,16 @@ public static void main(String[] args) {
...
+    public static String getRandomString(int length) {
+        StringBuffer sb = new StringBuffer();
+        String string = "i am bus test client!";
+        for (int i = 0; i < length; i++) {
+            int number = new Random().nextInt(string.length());
+            sb.append(string.charAt(number));
+        }
+        return sb.toString();
+    }
+
```
Comment (luchunliang) [inline inlong-sdk/dataproxy-sdk/src/main/java/org/apache/inlong/sdk/dataproxy/example/UdpClientExample.java:104]: Class "Random" is not secure. Please use SecureRandom and different seed.

Is getRandomString invalid code?

File: inlong-sdk/dataproxy-sdk/src/main/java/org/apache/inlong/sdk/dataproxy/codec/ProtocolEncoder.java [hunk 2]
```diff
@@ -287,8 +288,8 @@ private ByteBuf writeToBuf5(EncodeObject object) {
                         EncryptInfo encryptInfo = encryptEntry.getRsaEncryptInfo();
                         msgAttrs = msgAttrs + "_userName=" + object.getUserName()
                                 + "&_encyVersion=" + encryptInfo.getVersion()
-                                + "&_encyDesKey=" + encryptInfo.getRsaEncryptedKey();
-                        body = EncryptUtil.desEncrypt(body, encryptInfo.getDesKey());
+                                + "&_encyAesKey=" + encryptInfo.getRsaEncryptedKey();
+                        body = EncryptUtil.aesEncrypt(body, encryptInfo.getDesKey());
```
Comment (luchunliang) [inline inlong-sdk/dataproxy-sdk/src/main/java/org/apache/inlong/sdk/dataproxy/codec/ProtocolEncoder.java:292]: It is not fit that getDesKey is a AES Key.
body = EncryptUtil.aesEncrypt(body, encryptInfo.getDesKey());

File: inlong-sdk/dataproxy-sdk/src/main/java/org/apache/inlong/sdk/dataproxy/codec/ProtocolEncoder.java [hunk 3]
```diff
@@ -134,9 +135,9 @@ private ByteBuf constructBody(byte[] body, EncodeObject object,
                     }
                     EncryptInfo encryptInfo = encryptEntry.getRsaEncryptInfo();
                     endAttr = endAttr + "_userName=" + object.getUserName()
-                        + "&_encyVersion=" + encryptInfo.getVersion()
-                        + "&_encyDesKey=" + encryptInfo.getRsaEncryptedKey();
-                    body = EncryptUtil.desEncrypt(body, encryptInfo.getDesKey());
+                            + "&_encyVersion=" + encryptInfo.getVersion()
+                            + "&_encyAesKey=" + encryptInfo.getRsaEncryptedKey();
+                    body = EncryptUtil.aesEncrypt(body, encryptInfo.getDesKey());
```
Comment (luchunliang) [inline inlong-sdk/dataproxy-sdk/src/main/java/org/apache/inlong/sdk/dataproxy/codec/ProtocolEncoder.java:140]: It is not fit that getDesKey is a AES Key.
body = EncryptUtil.aesEncrypt(body, encryptInfo.getDesKey());

File: inlong-sdk/dataproxy-sdk/src/main/java/org/apache/inlong/sdk/dataproxy/config/EncryptConfigEntry.java [hunk 4]
```diff
@@ -99,20 +108,20 @@ public EncryptInfo getRsaEncryptInfo() {
...
-                this.desKey = EncryptUtil.generateDesKey();
+                this.aesKey = EncryptUtil.generateAesKey();
                 try {
-                    byte[] encryptedKey = EncryptUtil.rsaEncrypt(rsaKey, this.desKey);
+                    byte[] encryptedKey = EncryptUtil.rsaEncrypt(rsaKey, this.aesKey);
                     String tmpKey = Base64.encodeBase64String(encryptedKey);
                     rsaEncryptedKey = URLEncoder.encode(tmpKey, "utf8");
                     this.lastUpdateTime.set(System.currentTimeMillis());
-                    return new EncryptInfo(this.version, this.rsaEncryptedKey, this.desKey);
+                    return new EncryptInfo(this.version, this.rsaEncryptedKey, this.aesKey);
```
Comment (luchunliang) [inline inlong-sdk/dataproxy-sdk/src/main/java/org/apache/inlong/sdk/dataproxy/config/EncryptConfigEntry.java:117]: These codes is same with getRsaEncryptedKey.

Comment (woofyzhao) [reply inlong-sdk/dataproxy-sdk/src/main/java/org/apache/inlong/sdk/dataproxy/config/EncryptConfigEntry.java:117]: Looking into that in future PRs.

Comment (woofyzhao) [reply inlong-sdk/dataproxy-sdk/src/main/java/org/apache/inlong/sdk/dataproxy/config/EncryptConfigEntry.java:117]: Looking into the code structure problem in separate threads.

File: inlong-sdk/dataproxy-sdk/src/main/java/org/apache/inlong/sdk/dataproxy/example/UdpClientExample.java (hunk 1 above)
Comment (woofyzhao) [reply inlong-sdk/dataproxy-sdk/src/main/java/org/apache/inlong/sdk/dataproxy/example/UdpClientExample.java:106]: changed to SecureRandom

File: inlong-sdk/dataproxy-sdk/src/main/java/org/apache/inlong/sdk/dataproxy/example/UdpClientExample.java [hunk 5]
```diff
@@ -92,6 +92,16 @@ public static void main(String[] args) {
         }
     }
 
+    public static String getRandomString(int length) {
+        StringBuffer sb = new StringBuffer();
```
[the remark this answers is not in the document]
Comment (woofyzhao) [reply inlong-sdk/dataproxy-sdk/src/main/java/org/apache/inlong/sdk/dataproxy/example/UdpClientExample.java]: fixed

File: inlong-sdk/dataproxy-sdk/src/test/java/org/apache/inlong/sdk/dataproxy/EncryptUtilTest.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.nio.charset.StandardCharsets;
+
+public class EncryptUtilTest {
+
+    @Test
+    public void testEncryptDecrypt() {
+        String plainText = "TDB-30001 Create Tdw Table Error26880 FAILED: "
```
[the remark this answers is not in the document]
Comment (woofyzhao) [reply inlong-sdk/dataproxy-sdk/src/test/java/org/apache/inlong/sdk/dataproxy/EncryptUtilTest.java]: Changed to meaningful text.

**Review by luchunliang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by pocozh — APPROVED** [review]
LGTM

**Pull request merged**